### DISCUSSION TITLE
Scrolling animated background. Transformation layer doing my head in.

### DISCUSSION THREAD

**Discussion by michaelwhotoon** [keiwando/toonsquid#132] [Q&A]
Hi All

I have a 720 x 720 drawing on layer 1. Assign it to a transformation layer and it says the x position is 360. Shouldn't it be zero? So ok put in -360 to take it off left side of screen.

Duplicate layer 1 and assign to another transformation layer. Can't be done. I assign the second transforation layer to layer 1 duplication (lets call it layer 2) and the original transformation layer changes assignment to layer 2. I also have to assign the x position to something like +1060 to punt it off the right side of the screen. It should be -720 for the left  and +720 for the right.

PS I have no idea what I am doing and probably doing something wrong. I'm trying to translate a tutorial on callipeg
https://www.youtube.com/watch?v=dPg7GNHZc_k&t=66s

What is the best way of doing a scrolling background animation in Squidtoon. I can do it easily in Rough animator and Callipeg.

Thanks Legends.

**Comment by keiwando** [comment]
Hi!
For this scenario you don't need transform layers at all in ToonSquid. The pixel layers have [transform properties](https://toonsquid.com/handbook/transform/transform_properties/) themselves that you can animate using keyframes. Just switch into keyframing mode, make sure that [edit pixels](https://toonsquid.com/handbook/transform/transform_tool/#tool-options) is disabled and then drag the layers to where you want them to move to.

I made a quick screen recording to demonstrate that similar to the tutorial video:

https://user-images.githubusercontent.com/12989570/212926220-a5f4132d-360e-4da8-8482-d2b31026466a.MP4

If you want a perfect loop, you would then have to shorten the drawing to frame 8, since frame 9 is a repeat of frame 1. You can also increase the length of the drawing further and move the keyframes to make the movement slower and also reduce the fps, but I hope that the video is enough to just demonstrate the basics of position keyframing.

And [here's](https://toonsquid.com/handbook/transform/hierarchy/#coordinates) a short paragraph about the coordinate systems in ToonSquid.
